homeassistant/components/stream/hls.py

```diff
@@ -109,14 +137,67 @@ def name(self) -> str:
     @property
     def format(self) -> str:
         """Return container format."""
-        return "mpegts"
+        return "mp4"
 
     @property
     def audio_codec(self) -> str:
         """Return desired audio codec."""
         return "aac"
 
     @property
-    def video_codec(self) -> str:
-        """Return desired video codec."""
-        return "h264"
+    def video_codecs(self) -> tuple:
+        """Return desired video codecs."""
+        return {"hevc", "h264"}
+
+    @property
+    def container_options(self) -> dict:
+        """Return container options."""
+        return {"movflags": "empty_moov+default_base_moof"}
```

Comment:
The FFMPEG HLS muxer sets the following options when writing `fmp4` segments: https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/hlsenc.c#L866-L867

Any reason ours are different?

Reply:
'dash' wasn't documented, but if you have a look here https://github.com/FFmpeg/FFmpeg/blob/d5247fb1da615a7ccff9fa7271bc8aba23e0de25/libavformat/movenc.c#L6317 , it sets the same flags that we're using (empty moov+default_base_moof) plus the flag FF_MOV_FLAG_FRAGMENT. But FF_MOV_FLAG_FRAGMENT is redundant since it gets set here anyway: https://github.com/FFmpeg/FFmpeg/blob/d5247fb1da615a7ccff9fa7271bc8aba23e0de25/libavformat/movenc.c#L6311
frag_custom isn't necessary because that just specifies to cut the fmp4 manually, but since we are not using the fmp4 option this doesn't matter.
As for delay_moov, I didn't look into that. A quick google shows that it might keep the start time pegged at 0: https://stackoverflow.com/questions/44731531/how-to-keep-the-start-time-0-when-writing-a-mp4video-to-stdout-with-ffmpeg